crates/interfaces/src/db.rs

```diff
@@ -29,3 +31,58 @@ pub enum Error {
     #[error("Error decoding value.")]
     DecodeError,
 }
+
+// TODO: dedup with provider::Error?
```

Comment:
let's merge the two error types. would clean up some duplications

Reply:
`Error` and `DatabaseIntegrityError`?

both seem valuable on their own. And I'd assume `Error` is more low level. 
but no strong feelings otherwise.

Reply:
@mattsse this is the error that's mentioned
https://github.com/paradigmxyz/reth/blob/9470943babf8afe98a10db729954a374470afcb7/crates/interfaces/src/provider.rs#L3-L27